runtime/common/RuntimeMLIRCommonImpl.h

```diff
@@ -373,7 +375,16 @@ qirProfileTranslationFunction(const char *qirProfile, mlir::Operation *op,
     pm.enableStatistics();
   std::string errMsg;
   llvm::raw_string_ostream errOs(errMsg);
-  cudaq::opt::addPipelineConvertToQIR(pm, qirProfile);
+  bool containsWireSet =
+      op->walk<mlir::WalkOrder::PreOrder>([](quake::WireSetOp wireSetOp) {
+          return mlir::WalkResult::interrupt();
+        }).wasInterrupted();
```

Comment:
Won't this be more predictable (perhaps in the future)? It seems like we will know that if the target selected is _foo_ then the compiler will be running qubit management, mapping, etc. and scanning for the presence of an Op in the Module won't be needed or determinative of what has to happen next. If mapping uses wire_set and the output doesn't have a wire_set, then we can just raise a fatal error.